src/bundler.ts

```diff
@@ -20,32 +20,36 @@ import { AssemblerHooks } from './hooks.js'
 import type { BundlerOptions } from './types.js'
 import { run, parseConfig, copyFiles } from './helpers.js'
 
-type SupportedPackageManager = 'npm' | 'yarn' | 'pnpm' | 'bun'
+type SupportedPackageManager = 'npm' | 'yarn' | 'yarn@berry' | 'pnpm' | 'bun'
 
 /**
  * List of package managers we support in order to
  * copy lockfiles
  */
 const SUPPORT_PACKAGE_MANAGERS: {
   [K in SupportedPackageManager]: {
-    lockFile: string
+    packageManagerFiles: string[]
     installCommand: string
   }
 } = {
-  npm: {
-    lockFile: 'package-lock.json',
+  'npm': {
+    packageManagerFiles: ['package-lock.json'],
     installCommand: 'npm ci --omit="dev"',
   },
-  yarn: {
-    lockFile: 'yarn.lock',
+  'yarn': {
+    packageManagerFiles: ['yarn.lock'],
     installCommand: 'yarn install --production',
   },
-  pnpm: {
-    lockFile: 'pnpm-lock.yaml',
+  'yarn@berry': {
+    packageManagerFiles: ['yarn.lock', '.yarn/**/*', '.yarnrc.yml'],
+    installCommand: 'yarn workspaces focus --production',
```

Comment:
Sorry, not familiar with Yarn at all, so my question may be stupid

But, isn't `yarn workspaces` a thing for monorepos?

Reply:
Mmh finally, no, it seems that this is indeed the necessary command according to their migration guide: https://yarnpkg.com/migration/guide#cli-changes

Reply:
Yes, `yarn install --production` is now deprecated, when used it will just display a warning and install as if the argument was not present